### ISSUE TITLE
getSideMenu does not take child of child in to account

### ISSUE THREAD

**Issue by hvanoch** [sonata-project/SonataAdminBundle#5044]
Since the getSideMenu always returns the "upper" child of parent, there is no way to get the child of a child in the buildSideMenu.

I have an event entity which has multiple locations. Each location can have multiple lockers types in place.

I want to use the sidebar functionality to first show all the locations of an event. When they click the location I want to list the locker types at that location. The first part works perfectly. But how do I get the lockers list for the selected location?

The problem I have is that the getSideMenu method always uses the upper child. In this case the child of Event, so Location. And not the locker type. So I have no access to the locker type admin.

The configureSideMenu method of the EventAdmin looks like this:

```php
protected function configureSideMenu(MenuItemInterface $menu, $action, AdminInterface $childAdmin = null)
{
    if (!$childAdmin && !in_array($action, ['edit', 'show'])) {
        return;
    }

    $menu->addChild(
        'Event details',
        [
            'uri' => $this->generateUrl('show', ['id' => $this->getRequest()->get($this->getIdParameter())]),
        ]
    );


    $menu->addChild(
        'Location list',
        [
            'uri' => $this->generateUrl(LocationAdmin::class.'.list', ['id' => $this->getRequest()->get($this->getIdParameter())]),
        ]
    );

    $menu->addChild(
        'Locker type list',
        [
            'uri' => $childAdmin->generateUrl(
                LockerTypeAdmin::class.'.list',
                [
                    'id' => $this->getRequest()->get($childAdmin->getIdParameter()),
                ]
            ),
        ]
    );
}
```

Basically I want to show a list by clicking on the button of that parent entity like in the example https://sonata-project.org/bundles/admin/3-x/doc/reference/child_admin.html
